packages/datadog-plugin-next/src/index.js

```diff
@@ -1,149 +1,81 @@
 'use strict'
 
-// TODO: either instrument all or none of the render functions
-
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
 const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
 
-const contexts = new WeakMap()
-
-function createWrapHandleRequest (tracer, config) {
-  return function wrapHandleRequest (handleRequest) {
-    return function handleRequestWithTrace (req, res, pathname, query) {
-      return trace(tracer, config, req, res, () => handleRequest.apply(this, arguments))
-    }
+class NextPlugin extends Plugin {
+  static get name () {
+    return 'next'
   }
-}
-
-function createWrapHandleApiRequest (tracer, config) {
-  return function wrapHandleApiRequest (handleApiRequest) {
-    return function handleApiRequestWithTrace (req, res, pathname, query) {
-      return trace(tracer, config, req, res, () => {
-        const promise = handleApiRequest.apply(this, arguments)
-
-        return promise.then(handled => {
-          if (!handled) return handled
 
-          const page = getPageFromPath(pathname, this.dynamicRoutes)
-
-          addPage(req, page)
-
-          return handled
-        })
+  constructor (...args) {
+    super(...args)
+
+    this._requests = new WeakMap()
+
+    this.addSub('apm:next:request:start', ({ req, res }) => {
+      const store = storage.getStore()
+      const childOf = store ? store.span : store
+      const span = this.tracer.startSpan('next.request', {
+        childOf,
+        tags: {
+          'service.name': this.config.service || this.tracer._service,
+          'resource.name': 'test',
```

Comment:
FWIW I'm seeing this show up in my production traces

Reply:
Fixing now. However, the default value should rarely be seen since we add the API endpoint or the page to the resource name later on during the request. What were you seeing in these traces before? Was it only the HTTP method or the route to an API or a page?

Reply:
This is basically why our tests didn't fail, because the default value never reaches our tests.

Reply:
I’m not sure what it was like before, but all of the ones I am seeing now are for static URLs which should be static content — and probably not served via next, if that helps.

Reply:
Thanks for confirming, added a test for static files 👍